File: raspi_tank/src/tank.erl
-module(tank).
-export([init/2, set_speed/3, pi_blast/1]).

init(Left, Right) ->
  LeftMotor = motor:init(Left),
  RightMotor = motor:init(Right),
  {tank, LeftMotor, RightMotor}.

set_speed(Left, Right, {tank, LeftMotor, RightMotor}) ->
  {tank,
    motor:set_speed(Left, LeftMotor),
    motor:set_speed(Right, RightMotor)}.

pi_blast({tank, LeftMotor, RightMotor}) ->
  motor:pi_blast(LeftMotor),
  motor:pi_blast(RightMotor).

File: elixir/raspi_tank/lib/raspi_tank/tank.ex
defmodule RaspiTank.Tank do
  def init(device_string) do
    ArduinoMotorDriver.start(device_string)
  end

  def set_speed(left, right, device) do
    ArduinoMotorDriver.left(device, left)
    ArduinoMotorDriver.right(device, right)
  end

  def pi_blast({:tank, left_motor, right_motor}) do
    # LOL NOPE
  end
end
